### PR TITLE
feat: provide an option to force analysis

### LINKED ISSUE #45: Provide an option to force analysis
## Overview

**Is your feature request related to a problem? Please describe.**
I'm often frustrated when working with the `phylum-ci` entrypoint during test and evaluation scenarios and it tells me "The lockfile has not changed. Nothing to do." There are times when I want the analysis to run regardless.

**Describe the solution you'd like**
It would be great if there was an option to force analysis of the lockfile, even when it hasn't changed.

**Describe alternatives you've considered**
In order to "trigger" the analysis, I will often have to add or modify a dependency to the existing lockfile or create a new/different lockfile.

## Additional context

The option should default to False (not enabled) and only take effect if a user specifies the option. The option name could be something like `--force-analysis`, but suggestions for better names are welcome. Adding this feature will aid testing...local/manual and functional/integration.

## Acceptance criteria

- [x] An option is exposed to force analysis of the project lockfile
- [x] Documentation is updated

### PR DESCRIPTION
**Pull request by maxrake**
This PR adds an option to force analysis of the lockfile, even when it hasn't changed.

This PR also includes a fix to ensure the "CI Platform Name" portion of a label is correct. The order was wrong and all the labels were getting the default value of "Unknown". That fix can also be seen in the output below.

The PR will not be squashed...so that the proper tracking of the feat and fix will remain.

Closes #45

## Checklist

- [x] Does this PR have an associated issue (i.e., `closes #<issueNum>` in description above)?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [ ] ~Have you created sufficient tests?~
  - Still no automated tests; confirmed locally...see output below
- [x] Have you updated all affected documentation?

## Output

```
❯ poetry run phylum-ci -l poetry.lock --all-deps
 [+] No CI environment detected
 [+] Confirming pre-requisites ...
 [+] Existing `.phylum_project` file found at: /Users/coggins/dev/phylum/phylum-ci/.phylum_project
 [+] `git` binary found on the PATH
 [+] Existing `.git` directory was found at the current working directory
 [+] All pre-requisites met
 [+] lockfile in use: /Users/coggins/dev/phylum/phylum-ci/poetry.lock
 [+] The lockfile has not changed. Nothing to do.

❯ poetry run phylum-ci -l poetry.lock --all-deps -fa
 [+] No CI environment detected
 [+] Confirming pre-requisites ...
 [+] Existing `.phylum_project` file found at: /Users/coggins/dev/phylum/phylum-ci/.phylum_project
 [+] `git` binary found on the PATH
 [+] Existing `.git` directory was found at the current working directory
 [+] All pre-requisites met
 [+] lockfile in use: /Users/coggins/dev/phylum/phylum-ci/poetry.lock
 [+] `--force-analysis` flag specified. Proceeding with analysis ...
 [+] phylum_label: phylum-ci_No-CI_with_force_3a482fbcb2255c3e8784cf637a9cd5dc763d6bf6
 [+] Existing Phylum CLI instance found: v3.5.0 at /Users/coggins/.local/bin/phylum
 [+] Using Phylum CLI instance: v3.5.0 at /Users/coggins/.local/bin/phylum
 [*] Performing analysis ...
 [+] Considering all current dependencies ...
 [+] 61 current dependencies
 [!] The analysis is complete and there were failures
 [+] Project URL: https://app.phylum.io/projects/e6ddb53a-f600-405e-b8ab-e691a61a0c76?label=phylum-ci_No-CI_with_force_3a482fbcb2255c3e8784cf637a9cd5dc763d6bf6
 [-] Return code: 1
 [+] Analysis output:

## Phylum OSS Supply Chain Risk Analysis - FAILED
---TRIMMED---
```

### CODE REVIEW

**Comment by maxrake** [comment]
Adding @louislang since the rest of the integrations team is on leave or out sick.

**Review by louislang — APPROVED** [review]
lgtm!

File: src/phylum/ci/cli.py [hunk 1]
```diff
@@ -105,6 +105,12 @@ def get_args(args: Optional[Sequence[str]] = None) -> Tuple[argparse.Namespace,
         action="store_true",
         help="Specify this flag to consider all dependencies in analysis results instead of just the newly added ones.",
     )
+    analysis_group.add_argument(
+        "-fa",
```
Comment (cd-work) [inline src/phylum/ci/cli.py:109]: Unfortunately a bit late now, but I feel like this is not a good idea. Usually on conventional unix CLIs `-fa` would imply `-f -a`, thus it's not really appropriate to have two-letter shorthands?

A shorthand shouldn't be necessary, but if it is used it should probably be a single letter. Using lower/uppercase for different meanings is however possible.

Comment (maxrake) [reply src/phylum/ci/cli.py:109]: #56 has been created to track this issue.

**Pull request merged**